README
-*- Indented-Text -*-

$Id: README,v 4.36 2003/04/10 21:22:05 n0ll Exp $

======================================================================
				   
			F  I  D  O  G  A  T  E

			      Version 4

		     ============================
		        * Fido-Internet Gateway
		        * Fido FTN-FTN Gateway
		        * Fido Mail Processor
		        * Fido File Processor
		        * Fido Areafix/Filefix
		     ============================


WARNING!!! If you're migrating from FIDOGATE 4.2.9 to 4.4.x, please
note the following *MAJOR* changes:

     4.3a1..a5	New directory hierarchy, new config files
		/etc/fidogate/*.conf (see doc/dirs.html)

     4.3.0+	In addition, FIDOGATE now uses a single config file
		/etc/fidogate/fidogate.conf for *all* programs.

     4.4.x	The new stable release, even somewhat well documented.
		See ANNOUNCE for recent changes.


Please subscribe to the FIDOGATE mailing list!!!

======================================================================

[ For detailed instructions see doc/fidogate.txt ]

Needed:

* GNU CC or ANSI C compiler
* GNU make
* GNU flex/bison or lex/yacc
* Perl5
* sendmail m4 configuration files
* GNU m4 for the sendmail m4 configuration

Compiling:

1.  Edit config.make
2.  Edit config.h if needed
3.  Run "make depend"
4.  Run "make"

Installing:

5.  Run "make install-dirs" to create the installation directories
    (only if needed)
6.  Run "make install" to install the programs
7.  Create the various config files and put them into CONFIGDIR
    (see examples/... to get started)

See also doc/fidogate.txt (PostScript documentation can be compiled by
running "make doc" in the doc/ subdirectory) for a detailed
description of the FIDOGATE installation.

======================================================================

There is a *NEW* mailing list for FIDOGATE discussions and
announcements:

    mailto:fidogate-list@lists.sourceforge.net

To SUBSCRIBE to this mailing list goto

    http://lists.sourceforge.net/lists/listinfo/fidogate-list

======================================================================

Latest version:

The latest version of FIDOGATE can be found at:

    http://sourceforge.net/project/showfiles.php?group_id=10739

    ftp://ftp.fidogate.org/pub/fidogate/
    http://www.fidogate.org/pub/fidogate/


See also the FIDOGATE home page at:

    http://www.fidogate.org/
    http://fidogate.sourceforge.net/

======================================================================

Warning: configuration changed a lot since release 3.9.7, so using an
old configuration without any changes for release 4 will cause
problems.

In particular, Roland's fgpack script (see doc/Fido-Point-HOWTO) does
*NOT* work with the release 4 ftnpack!!!

======================================================================

GNU-WIN32 version:

As of FIDOGATE 4.2.5, the source can be compiled with the Cygnus
GNU-WIN32 development environment for Windows 95 and NT. You'll need
the WIN32 version of perl, too. See examples/win32/config.make.

GNU-WIN32 information can be found at
    http://www.cygnus.com/misc/gnu-win32/

Perl for WIN32 information can be found at
    http://www.activestate.com/

======================================================================

Contributing patches:

PLEASE NOTE THE FOLLOWING RULES FOR CONTRIBUTING PATCHES:

* No changes to the indentation, white space etc. No beautifying,
  please.

* Supply only context diffs (command diff -u), not complete changed
  files.

* Supply patches against the latest version of FIDOGATE, or even
  better against the FIDOGATE CVS repository.

If you do not follow these simple rules, it is quite likely that I'll
not integrate your patches.

======================================================================

$Date: 2003/04/10 21:22:05 $
Martin Junius
mailto:mj@fido.de


ChangeLog
2016-03-05  gettextize  <bug-gnu-gettext@gnu.org>

	* Makefile.am
	(ACLOCAL_AMFLAGS): New variable.
	(EXTRA_DIST): New variable.

2008-11-22  Andrey Slusar  <anrays@gmail.com>

	* src/common/binkley.c:
	* src/toss/ftnpack.c:
	* src/toss/ftnroute.c:
	* acconfig.h:
	* configure.in:
	* doc/examples/dyff/fidogate.conf.sample.in:
	* doc/FAQ_Config.ru:
	* doc/README.ru:
	In fidogate.conf added option AmigaStyleOutbound - use ASO instead
	BSO and removed configure option `--enable-amiga-out'(patch from
	Yauheni Kaliuta).

2008-10-01  Andrey Slusar  <anrays@gmail.com>

	* configure.in: Fix `Unrecognized options' warnings.

2008-09-30  Andrey Slusar  <anrays@gmail.com>

	* src/areafix/Makefile.am:
	* src/charset/Makefile.am:
	* src/gate/Makefile.am:
	* src/toss/Makefile.am:
	Change `install-exec-am' to `installcheck-am' for fix install.

2008-09-21  Andrey Slusar  <anrays@gmail.com>

	* src/include/prototypes.h:
	* src/common/textlist.c(tl_for_each): New function tl_for_each()
	(patch from Yauheni Kaliuta).
	* src/gate/ftn2rfc.c: A hack to encode all non-7bit headers
	(patch from Yauheni Kaliuta).
	* acinclude.m4: Update to new aclocal(tested on 1.10.1).
	* src/toss/Makefile.am: Remove runinc individual CFLAGS.
	* src/charset/Makefile.am: Update to new automake(tested on 1.10.1).
	* autogen.sh: Add script for build configure, Makefiles and
	others autotool-files.

2008-09-17  Andrey Slusar  <anrays@gmail.com>

	======= Release 5.2.4ds ========================
	------- SVN tag R524 ---------------------------
	* packages/freebsd/Makefile:
	* packages/rpm/fidogate.spec:
	* Changes.ru:
	* version.h:
	* configure.in: Update to 5.2.4

2008-06-29  Andrey Slusar  <anrays@gmail.com>

	* configure:
	* configure.in:
	Fix option `--enable-ai1'(thanks to Max N. Boyarov).

2008-06-24  Max N. Boyarov  <m.boyarov@bsd.by>

	* src/areafix/areafix.c (cmd_new): Fix
	another initialization to avoid segfault
	on free(3)

2007-12-21  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c:
	* src/gate/ftn2rfc.c:
	* src/include/prototypes.h:
	Fix type for buffer(Thanks to Zhenya Kaliuta).

2007-10-13  Andrey Slusar  <anrays@gmail.com>

	* src/toss/runinc.c (unpacking_parse_line): Fix
	unarc check when HAVE_STRICMP defined(reported by
	Alex Kocharin).

2007-09-12  Andrey Slusar  <anrays@gmail.com>

	* src/gate/ftn2rfc.c (snd_message): Add ':' to
	REPLYADDR/REPLYTO when ReplyAddrIfmailTX enable.

2007-08-29  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c (mime_enheader):
	(mime_deheader):
	* src/include/prototypes.h (mime_enheader):
	Correct types for variables.

2007-04-07  Andrey Slusar  <anrays@gmail.com>

	* src/toss/ftntoss.c (do_unknown_area): Correct
	works `KillUnknown' option(thanks to Dmitry Rodin).

2007-01-30  Andrey Slusar  <anrays@gmail.com>                                   

	* src/gate/rfc2ftn.c (snd_message): Fix works
	option `ReplyAddrIfmailTX'.

2007-01-29  Andrey Slusar  <anrays@gmail.com>

	* src/toss/ftntoss.c (do_echomail): Fix tossing,
	when packet in passwd defined.

2007-01-17  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c: Add mime allocation function.
	* src/include/structs.h (st_mimeinfo): Add next.
	* src/common/hosts.c (hosts_free):
	* src/common/passwd.c (passwd_free):
	Remove unneeded if.
	* src/util/ftnoutpkt.c (do_addr):
	* src/toss/runinc.c (run_toss):
	free()->xfree() for more portable.

2007-01-16  Andrey Slusar  <anrays@gmail.com>

	* src/common/tmps.c: More comments in fatal() messages.
	* src/common/mime.c: Fix crashes.

2007-01-10  Andrey Slusar  <anrays@gmail.com>

	* src/include/fidogate.h: Different INTERNAL_CHARSET and
	INTERNAL_TYPE when defined and not defined HAVE_ICONV.
	(thanks to Zhenya Kaliuta)
	* src/gate/rfc2ftn.c: Fix recoding when HAVE_ICONV defined.
	(thanks to Zhenya Kaliuta)

2007-01-09  Andrey Slusar  <anrays@gmail.com>

	* scripts/outbound/outb-kill.in: Fix ftnflo location
	(Thanks to Yuri Chumakov)

2007-01-08  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c:
	* src/gate/rfc2ftn.c:
	Fix 8-bit subject handling.
	If subject was not mimed, recode it according to message
	charset(patch from Zhenya Kaliuta).

2007-01-06  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c (mime_debody_section): Fix works when
	mime type or encoding is empty.

2006-12-30  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c: Add changes for HAVE_ICONV(fix
	previous commit).

2006-12-28  Andrey Slusar  <anrays@gmail.com>

	* Changes.ru:
	* doc/README.ru:
	* doc/examples/dyff/fidogate.conf.sample.in:
	* doc/examples/point/fidogate.conf.in:
	* doc/examples/snp/fidogate.conf:
	Removed AutoCreateDontCheckPassword and updated Changes for
	this.
	* src/common/areasbbs.c (areasbbs_free): xfree() without
	checking.
	* src/common/mime.c:
	* src/common/rfcheader.c:
	* src/common/textlist.c:
	* src/include/fidogate.h:
	* src/include/prototypes.h:
	* src/include/structs.h:
	* acconfig.h:
	Added conversations with iconv support and fix recoding(patch
	from Zhenya Kaliuta).

2006-12-27  Andrey Slusar  <anrays@gmail.com>

	* src/toss/ftntoss.c (do_echomail): Change password
	checking in echomail packets - when password is incorrect,
	packet moved to bads.
	* src/toss/ngoper.c: Add cvs Id.

2006-11-15  Andrey Slusar  <anrays@gmail.com>

	* src/areafix/areafix.c (cmd_sub): Fix potential
	crashes.
	* src/tick/ftntick.c (process_tick): Fix crashes
	(bug, reported by Eugene V. Subbotin)

2006-11-05  Andrey Slusar  <anrays@gmail.com>

	* src/common/mime.c:
	* src/gate/ftn2rfc.c:
	* src/gate/rfc2ftn.c:
	* src/include/fidogate.h:
	* src/include/prototypes.h:
	Add decoding bodyes in base64(patch from Zhenja Kaliuta).

2006-10-31  Andrey Slusar  <anrays@gmail.com>

	* src/common/Makefile.am:
	* src/common/areas.c:
	* src/common/mime.c:
	* src/common/xstrnlen.c(new file):
	* src/gate/ftn2rfc.c:
	* src/include/prototypes.h:
	* src/include/structs.h:
	Add and implement option NetMailHeadersBase64, fix 
	decode headers(patch from Zhenja Kaliuta).


acconfig.h

/* Name of this package (needed by automake) */
#undef PACKAGE

/* Version of this package (needed by automake) */
#undef VERSION

#undef DEFAULT_MAILER

/* Define if you have the gettimeofday function.  */
#undef HAVE_GETTIMEOFDAY

/* Define if you have the link function.  */
#undef HAVE_LINK

/* Define if you have the mkdir function.  */
#undef HAVE_MKDIR

/* Define if you have the mktime function.  */
#undef HAVE_MKTIME

/* Define if you have the sprintf function.  */
#undef HAVE_SPRINTF

/* Define if you have the statfs function.  */
#undef HAVE_STATFS

/* Define if you have the strcasecmp function.  */
#undef HAVE_STRCASECMP

/* Define if you have the strerror function.  */
#undef HAVE_STRERROR

/* Define if you have the stricmp function.  */
#undef HAVE_STRICMP

/* Define if you have the strspn function.  */
#undef HAVE_STRSPN

/* Define if you have the strstr function.  */
#undef HAVE_STRSTR

/* Define if you have the syslog function.  */
#undef HAVE_SYSLOG

/* Define if you have the <dirent.h> header file.  */
#undef HAVE_DIRENT_H

/* Define if you have the <fcntl.h> header file.  */
#undef HAVE_FCNTL_H

/* Define if you have the <limits.h> header file.  */
#undef HAVE_LIMITS_H

/* Define if you have the <malloc.h> header file.  */
#undef HAVE_MALLOC_H

/* Define if you have the <ndir.h> header file.  */
#undef HAVE_NDIR_H

/* Define if you have the <sys/dir.h> header file.  */
#undef HAVE_SYS_DIR_H

/* Define if you have the <sys/mount.h> header file.  */
#undef HAVE_SYS_MOUNT_H

/* Define if you have the <sys/ndir.h> header file.  */
#undef HAVE_SYS_NDIR_H

/* Define if you have the <sys/param.h> header file.  */
#undef HAVE_SYS_PARAM_H

/* Define if you have the <sys/statfs.h> header file.  */
#undef HAVE_SYS_STATFS_H

/* Define if you have the <sys/statvfs.h> header file.  */
#undef HAVE_SYS_STATVFS_H

/* Define if you have the <sys/time.h> header file.  */
#undef HAVE_SYS_TIME_H

/* Define if you have the <sys/vfs.h> header file.  */
#undef HAVE_SYS_VFS_H

/* Define if you have the <sysexits.h> header file.  */
#undef HAVE_SYSEXITS_H

/* Define if you have the <syslog.h> header file.  */
#undef HAVE_SYSLOG_H

/* Define if you have the <time.h> header file.  */
#undef HAVE_TIME_H

/* Define if you have the <unistd.h> header file.  */
#undef HAVE_UNISTD_H


/***** General configuration *************************************************/

/*
 * Generate local FTN addresses, e.g.
 *     user_name%p.f.n.z@host.domain
 * instead of
 *     user_name@p.f.n.z.domain
 */
#undef LOCAL_FTN_ADDRESSES

/*
 * Create Binkley-style BSY files for all outbound operations
 */
#undef DO_BSY_FILES

/*
 * Create lock files / BSY files in an NFS-safe way (see man 2 open)
 */
#undef NFS_SAFE_LOCK_FILES

/*
 * Default max. message size for FIDO. Due to some more brain damage
 * in FIDONET programs we have to split larger messages into several
 * smaller ones. May be set with the -M option in AREAS or MaxMsgSize
 * in DEFAULT_CONFIG_GATE.
 */
/* < 16 K */
/* #define MAXMSGSIZE	14000 */
/* < 32 K */
/* #define MAXMSGSIZE	30000 */
#undef MAXMSGSIZE
/*
 * Domain for invalid FTN addresses
 */
#undef FTN_INVALID_DOMAIN

/*
 * syslog facility used for logging if logfile == "syslog"
 * (only for HAS_SYSLOG defined)
 */
#undef FACILITY

/*
 * Default assumed charset for Fido messages (see also DefaultCharset)
 */
#undef CHARSET_STDFTN

/*
 * Default assumed charset for RFC messages if without MIME headers
 */
#undef CHARSET_STDRFC

/*
 * Default charset for RFC messages with forced 7bit encoding
 */
#undef CHARSET_STD7BIT


/***** ftn2rfc configuration ************************************************/

/*
 * Rewrite addresses found in ALIASES so that the sender's address is the
 * gateway address. The reverse direction requires suitable MTA aliases.
 */
#undef ALIASES_ARE_LOCAL

/***** rfc2ftn configuration ************************************************/

/** Passthru operation for NetMail: FIDO->Internet->FIDO **/
#undef PASSTHRU_NETMAIL
/** Passthru operation for EchoMail: FIDO->Internet->FIDO **/
/*
 * Implemented, but requires ftntoss run after rfc2ftn to sort SEEN-BY
 */
#undef PASSTHRU_ECHOMAIL

/** Support newsserver SN ***************************************************/
/*
 * Enable SN-style active_lookup
 */
#undef SN

/***** AI patches configuration (see README.ai) *****************************/
/*
 * Add -a option to HOSTS entries, useful only with PASSTHRU_NET/ECHOMAIL
 */
#undef AI_1

/*
 * Enable best AKA select
 */
#undef BEST_AKA

/*
 * Enable ACL for FTN links
 */
#undef FTN_ACL

/*
 * Put file descriptions into `.desc/%s.desc' file instead of files.bbs
 */
 // .../fileecho/.desc/file.zip.desc
// #define DESC_DIR	".desc"
// .../fileecho/file.zip.desc
// #define DESC_DIR	""

#undef DESC_DIR

/*
 * Add `X-FTN-From' header in gated echomail.
 */
#undef X_FTN_FROM_ECHOMAIL

/*
 * Added the feature "spy forwards".
 */
#undef SPYES

/*
 * Restricted access to areafix command `%listall'
 */
#undef AF_LISTALL_RESTRICTED

/*
 * Recode file descriptions
 */
#undef RECODE_FILE_DESC

/*
 * Enable areafix command `%avail', change behaviour of areafix
 * command `%list'
 */
#undef AF_AVAIL

/*
 * Enable passthrought filearea
 */
#undef FECHO_PASSTHROUGHT

/*
 * Enable if use fileboxes for fileecho
 */
#undef USE_FILEBOX

/*
 * Enable if no ftntick CRC control
 */
#undef FTNTICK_NOCRC

/*
 * Enable if no insert Origin line if it absence
 */
#undef NOINSERT_ORGANIZATION

/*
 * If Origin line is not present, include link address
 */
#undef INSERT_ORIGIN

/*
 * Don't put insecure mail into dupe database
 */
#undef INSECURE_DONT_PUT_INTO_DUPE_DB

/*
 * For the security reason enabled additional addresses checking in the  
 * .pkt header, message header, origin and echomail ^aPATH.
 * If the last address in PATH is not equal to address in the pkt header, 
 * then the address from the pkt header is inserted into PATH.
 */
#define SECURITY

/*
 * Do'nt put info about create to log if echo created by forward reques.
 */
#undef CREATE_LOG_FORWREQ

/*
 * Change ^AVia format from:
 *   ^AVia FIDOGATE/ftntoss 2:5030/1229.0, 20010405.164323.MSK
 * to:
 *   ^AVia 2:5030/1229.0 @20010405.164323.MSK FIDOGATE/ftntoss
 */
#define FTS_VIA

/*
 * Change areafix answer when subscibing to area from `subscribe' to `ok'.
 */
#undef ANSWER_OK

/*
 * Set fido-style msgid
 */
#undef FIDO_STYLE_MSGID

/*
 * Set msgid correspond data base
 */
#undef DBC_HISTORY

/*
 * Delete MIME if rfc status set to 2
 */
#undef DEL_MIME_IF_RFC2

/*
 * Enable areafix send echo status
 */
#undef AFSEND_ECHO_STATUS

/*
 * Enable subscribe autorization only if node listed in zonegate
 */
#undef SUBSCRIBE_ZONEGATE

/*
 * Paste domain name to Origin message line
 */
#undef DOMAIN_TO_ORIGIN

/*
 * Enable paste ^PID (mail or news reader name) and ^TID (fidogate version)
 * kludge
 */
#undef PID_READER_TID_GTV

/*
 * Enable tic dupe check
 */
#undef TIC_HISTORY

/*
 * Disable tosting netmail and set directory for incoming and outgoing pkt's
 */
#undef DO_NOT_TOSS_NETMAIL 

/*
 * Ignore `From:' field if REPLYTO or REPLYADDR was found
 */
#undef IGNORE_FROM_IF_REPLY

/*
 * Separate outouing message to individual pkt file
 */
#undef SEP_RFC2FTN_PKT

/*
 * Enable set limit for subscribe areas
 */
#define SUB_LIMIT

/*
 * Enable overwrite hardlink if file already exist
 */
#undef OVERWRITEN_FECHO_FILE_TO_LINK

/*
 * Enable lookup to INN active file
 */
#undef ACTIVE_LOOKUP

/*
 * Recode all RFC headers
 */
#define RECODE_ALL_RFC

/*
 * Enable use old batcher in runinc program
 */
#undef OLD_BATCHER

/*
 * Enable use send_fidogate function in runinc program
 */
#undef USE_RUNINC_SFGT

/*
 * Enable if you want use syslog for log messages
 */
#undef USE_SYSLOG

/*
 * Enable old style tosting
 */
#undef OLD_TOSS

/*
 * Fix bad year in pkt's
 */
#undef FIX_BAD_PKT_YEAR

/***** System dependend configuration ***************************************
 *
 *   HAVE_FCNTL_LOCK		Do you have file locking with fcntl()
 *
 *   HAVE_GETTIMEOFDAY	        Do you have gettimeofday()?
 *
 *   HAVE_TM_GMTOFF		Does your (struct tm) have a tm_gmtoff field?
 *
 *   HAVE_SYSEXITS_H             Do you have sysexits.h?
 *
 *   HAVE_TM_ZONE		Does your (struct tm) have a tm_zone field?
 *
 *   HAVE_STRFTIME		Do you have strftime()?
 *
 *   HAVE_TZNAME			Do you have extern char *tzname[2]?
 *
 *   HAVE_STRCASECMP		Do you have strcasecmp(), strncasecmp()?
 *
 *   HAVE_STRICMP		Do you have stricmp(), strnicmp()?
 *
 *
 *   HAVE_STRERROR               Do you have strerror()?
 *
 *
 *   DO_BINARY			Open files in binary mode
 *
 *   DO_DOSIFY			DOSify program names for execution
 *
 *
 *   RECEIVED_BY_MAILER "Received: by NeXT.Mailer"
 *				Define this if your mail system always
 *				generates something like
 *				"Received: by NeXT.Mailer"
 *
 *   HAVE_SPRINTF               sprintf(), vsprintf() supported
 *
 *   HAVE_SYSLOG		syslogd, syslog(), vsyslog() supported
 *
 *   HAVE_HARDLINKS		hardlinks supported by link() and filesystem
 *
 *   HAS_POSIX_REGEX		POSIX regcomp(), regexec() etc. supported
 *
 *   HAVE_SYS_VFS_H		Do you have statfs() in sys/vfs.h
 *
 *   HAVE_SYS_STATFS_H		Do you have statfs() in sys/statfs.h
 *
 *   HAVE_SYS_STATVFS_H		Do you have statfs() in sys/statvfs.h
 *
 *   HAVE_SYS_MOUNT_H		Do you have statfs() in sys/mount.h
 */

/***** Unix and derivates ***************************************************/

# undef  HAVE_FCNTL_H
# undef  HAVE_GETTIMEOFDAY
# undef  HAVE_SYSEXITS_H
# undef  HAVE_STRCASECMP
# undef  HAVE_STRICMP
# undef  HAVE_STRERROR
# undef  HAVE_SPRINTF
# undef  HAVE_TM_GMTOFF
# undef  HAVE_STRFTIME
# undef  HAVE_TM_ZONE
# undef  HAVE_TZNAME
#ifdef USE_SYSLOG
# undef  HAVE_SYSLOG
#endif
# undef  HAVE_SYS_MOUNT_H
# undef  HAVE_SYS_PARAM_H
#if !defined(HAVE_SYS_MOUNT_H) && !defined(HAVE_SYS_PARAM_H)
# undef  HAVE_SYS_VFS_H
# undef  HAVE_SYS_STATFS_H
# undef  HAVE_SYS_STATVFS_H
#endif
# undef  HAVE_TIME_H
# undef  HAVE_SYS_TIME_H

/* Standard config: POSIX UNIX */
# undef HAVE_LINK

/***** (MSDOS, OS/2, Windows) ************************************************/
#ifdef MSDOS			/* MSDOS, DJGPP GNU gcc */
#define  DO_DOSIFY
#endif
#ifdef __EMX__			/* OS/2, EMX GNU gcc */
#define  DO_DOSIFY
#endif
#ifdef __CYGWIN32__		/* GNU-Win32 Beta 20.1 */
#define  DO_DOSIFY
#endif

/* Reset some #define's based on system config */
#ifndef  HAVE_LINK
# undef  NFS_SAFE_LOCK_FILES
#endif

/***** End of configuration *************************************************/



/***** ^AMSGID/Message-ID configuration *************************************/

/*
 * Standard FIDONET domain for Z1-6 Message-IDs
 * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 * !!! DON'T TOUCH THIS, IF YOU'RE NOT ABSOLUTELY SURE WHAT YOU'RE DOING !!!
 * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 */
#define MSGID_FIDONET_DOMAIN	".fidonet.org"



/****************************************************************************
 *                                                                          *
 *              Think twice before changing anything below!!!               *
 *                                                                          *
 ****************************************************************************/

/*
 * Permissions
 */
#define PACKET_MODE	0600		/* Mode for outbound packets */
#define BSY_MODE	0644		/* Mode for BSY files */
#define FLO_MODE	0644		/* Mode for FLO files */
#define DIR_MODE	0755		/* Mode for directories */
#define FILE_DIR_MODE	0700		/* Mode for create fileecho directories */
#define CONF_MODE	0640		/* Mode for written config files */

/*
 * RFC headers recognized at beginning of FTN message body
 */
#define FTN_RFC_HEADERS \
    "From:", "Reply-To:", "UUCPFROM:", "To:", "Cc:", "Bcc:", \
    "Newsgroups:", "Sender:", "Content-Transfer-Encoding:", \
    "Header-To:", "Header-Cc:", "Subject:", "User-Agent:", \
    "X-Mailer:", "X-Newsreader:", "In-Reply-To:", "References:"

/*
 * RFC headers output for ^ARFC level 1 (partial RFC headers)
 */
#define RFC_LVL_1_HEADERS \
    "Reply-To", "Message-ID", "References"

/*
 * RFC headers output for ^ARFC level 3 (all RFC headers, excluding some)
 */
#define RFC_LVL_3_HEADERS \
    "NNTP-Posting-Date", \
    "NNTP-Posting-Host", \
    "X-Trace", \
    "X-Complaints-To", \
    "Path", \
    "Cache-Post-Path", \
    "X-Cache", \
    "Distribution", \
    "Received", \
    "Supersedes", \
    "X-Flags", \
    "X-MimeOLE", \
    "X-Priority", \
    "X-MSMail-Priority", \
    "X-Accept-Language", \
    "X-Mailer", \
    "X-Newsreader", \
    "X-Gateway", \
    "User-Agent", \
    "X-GateSoftware", \
    "In-Reply-To", \
    "From", \
    "Reply-To", \
    "Sender", \
    "To", \
    "CC", \
    "X-Comment-To", \
    "Newsgroups", \
    "Subject", \
    "Date", \
    "Organization", \
    "Lines", \
    "Message-ID", \
    "References", \
    "Mime-Version", \
    "Content-Type", \
    "Content-Transfer-Encoding", \
    "Xref", \
    "X-Original-Date", \
    "X-FTN-Kludge", \
    "X-FTN-Tearline", \
    "X-FTN-Origin", \
    "X-FTN-Seen-By", \
    "X-FTN-Path"

/*
 * Open modes for fopen(), binary for system requiring this.
 */
#ifdef DO_BINARY
# define R_MODE		"rb"
# define W_MODE		"wb"
# define A_MODE		"ab"
# define RP_MODE	"r+b"
# define WP_MODE	"w+b"
# define AP_MODE	"a+b"
#else
# define R_MODE		"r"
# define W_MODE		"w"
# define A_MODE		"a"
# define RP_MODE	"r+"
# define WP_MODE	"w+"
# define AP_MODE	"a+"
#endif /**DO_BINARY**/
#define R_MODE_T	"r"
#define W_MODE_T	"w"
#define A_MODE_T	"a"
#define RP_MODE_T	"r+"
#define WP_MODE_T	"w+"
#define AP_MODE_T	"a+"

/*
 * Format strings for dates
 */
#define DATE_LOG	"%b %d %H:%M:%S"

#define DATE_DEFAULT	"%a, %d %b %Y %H:%M:%S %O"
#define DATE_NEWS	"%a, %d %b %Y %H:%M:%S %O"
#define DATE_MAIL	"%a, %d %b %Y %H:%M:%S %O (%Z)"
#define DATE_FROM	"%a %b %d %H:%M:%S %Y"

#define DATE_FTS_0001	"%d %b %y  %H:%M:%S"
#define DATE_TICK_PATH	"%a %b %d %H:%M:%S %Y %Z"
#ifndef FTS_VIA
#define DATE_VIA	"%a %b %d %Y at %H:%M:%S %Z"
#else
//
//	%a	Abbr. weekly
//	%b	Abbr. Month
//	%d	Day of month
//	%H	Hour (local 24h)
//	%G	Hour (UTC 24h)
//	%V	Minute (UTC)
//	%U	"UTC" word
//	%j	Day of year
//	%m	Month
//	%M	Minutes (local)
//	%S	Second
//	%w	Day of week
//	%x	Date (Mon DD YYYY)
//	%X	Time (HH:MM:SS)
//	%y	Year (00-99)
//	%Y	Year (1900...)
//	%Z	Timezone (example: MSK)
//	%N	Time diff to TZUTC (if diff +03:30 - "+0330")
//	%O	Time diff to UTC (if diff +03:30 - "0330")
#define DATE_VIA	"%Y%m%d.%H%M%S.%N"
#endif /* FTS_VIA */
#define DATE_SPLIT	"%d %b %y %H:%M:%S"

/*
 * Product code for packets generated by FIDOGATE, 0xfe is used because
 * this code is reserved for new products, when the code numbers ran out.
 */
#define PRODUCT_CODE	0xfe

/*
 * Hard limits compiled into FIDOGATE
 */
#define MAXADDRESS	32		/* Max. # of FTN address in CONFIG */

#undef AREAFIXMAXSTR

#define MAXMSGHEADRLEN	70		/* Max. length message field */

#define MAXDOSDRIVE	16		/* Max. # of DOS drives in CONFIG */

#ifndef MAXPATH				/* Already defined by DJGPP */
# define MAXPATH	128		/* Max. size of path names */
#endif

#define MAXINETADDR	128		/* Max. size of an Internet address */

#define MAXUSERNAME	128		/* Max. size of an user name */

#define MAXOPENFILES	10		/* Max. # of open packet files used
					 * by ftntoss/ftnroute, this value
					 * should work on all supported
					 * systems, it can be incremented with
					 * ftntoss/ftnroute's -M option */

/* Define if using iconv */
#undef HAVE_ICONV

/* Define if using alloca.c.  */
#undef C_ALLOCA

/* Define to empty if the keyword does not work.  */
#undef const

/* Define to one of _getb67, GETB67, getb67 for Cray-2 and Cray-YMP systems.
   This function is required for alloca.c support on those systems.  */
#undef CRAY_STACKSEG_END

/* Define if you have alloca, as a function or macro.  */
#undef HAVE_ALLOCA

/* Define if you have <alloca.h> and it should be used (not on Ultrix).  */
#undef HAVE_ALLOCA_H

/* Define if you don't have vprintf but do have _doprnt.  */
#undef HAVE_DOPRNT

/* Define if you have the strftime function.  */
#undef HAVE_STRFTIME

/* Define if you have <sys/wait.h> that is POSIX.1 compatible.  */
#undef HAVE_SYS_WAIT_H

/* Define if your struct tm has tm_zone.  */
#undef HAVE_TM_ZONE

/* Define if you don't have tm_zone but do have the external array
   tzname.  */
#undef HAVE_TZNAME

/* Define if utime(file, NULL) sets file's timestamp to the present.  */
#undef HAVE_UTIME_NULL

/* Define if you have the vprintf function.  */
#undef HAVE_VPRINTF

/* Define if your C compiler doesn't accept -c and -o together.  */
#undef NO_MINUS_C_MINUS_O

/* Define to `long' if <sys/types.h> doesn't define.  */
#undef off_t

/* Define to `int' if <sys/types.h> doesn't define.  */
#undef pid_t

/* Define as the return type of signal handlers (int or void).  */
#undef RETSIGTYPE

/* Define to `unsigned' if <sys/types.h> doesn't define.  */
#undef size_t

/* If using the C implementation of alloca, define if you know the
   direction of stack growth for your system; otherwise it will be
   automatically deduced at run-time.
 STACK_DIRECTION > 0 => grows toward higher addresses
 STACK_DIRECTION < 0 => grows toward lower addresses
 STACK_DIRECTION = 0 => direction of growth unknown
 */
#undef STACK_DIRECTION

/* Define if you have the ANSI C header files.  */
#undef STDC_HEADERS

/* Define if you can safely include both <sys/time.h> and <time.h>.  */
#undef TIME_WITH_SYS_TIME

/* Define if your <sys/time.h> declares struct tm.  */
#undef TM_IN_SYS_TIME

#undef DEFAULT_MAILER

#undef HAVE_DLD

#undef HAVE_LIBDL

#undef HAVE_SHL_LOAD
